benches/src/erc20.rs

```diff
@@ -0,0 +1,194 @@
+use alloy::{
+    network::{AnyNetwork, EthereumWallet},
+    primitives::Address,
+    providers::ProviderBuilder,
+    sol,
+    sol_types::SolConstructor,
+    uint,
+};
+use alloy_primitives::U256;
+use e2e::{receipt, Account};
+use koba::config::Deploy;
+
+use crate::ArbOtherFields;
+
+const RPC_URL: &str = "http://localhost:8547";
+
+sol!(
+    #[sol(rpc)]
+    contract Erc20 {
+        function name() external view returns (string name);
+        function symbol() external view returns (string symbol);
+        function decimals() external view returns (uint8 decimals);
+        function totalSupply() external view returns (uint256 totalSupply);
+        function balanceOf(address account) external view returns (uint256 balance);
+        function allowance(address owner, address spender) external view returns (uint256 allowance);
+
+        function cap() public view virtual returns (uint256 cap);
+
+        function mint(address account, uint256 amount) external;
+        function burn(uint256 amount) external;
+        function burnFrom(address account, uint256 amount) external;
+
+        function transfer(address recipient, uint256 amount) external returns (bool);
+        function approve(address spender, uint256 amount) external returns (bool);
+        function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+    }
+);
+
+sol!("../examples/erc20/src/constructor.sol");
+
+const TOKEN_NAME: &str = "Test Token";
+const TOKEN_SYMBOL: &str = "TTK";
+const CAP: U256 = uint!(1_000_000_U256);
+
+pub async fn bench() -> eyre::Result<()> {
+    let alice = Account::new().await?;
+    let alice_addr = alice.address();
+    let alice_wallet = ProviderBuilder::new()
+        .network::<AnyNetwork>()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(alice.signer.clone()))
+        .on_http(alice.url().parse()?);
+
+    let bob = Account::new().await?;
```

Comment:
How about using `AccountFactory` from E2E tests?

Reply:
Hmmm, what do you mean? This is using the `AccountFactory` under the hood.

Reply:
Oh, you are right, never mind.